app/client/index.tsx

```diff
@@ -49,21 +61,40 @@ const sidebarRoutes = [
     }
 ];
 
-class Root extends React.Component<{}, {}> {
+class Root extends React.Component<{}, {signedIn: boolean}> {
+    private static didAttachListener: boolean = false;
+
     constructor() {
         super();
+        this.state = {
```

Comment:
I think this can just be shortened to this.state = { signedIn: CONFIG.getLongName().isJust() }